libzecale/circuits/fields/fp6_3over2_gadgets.hpp

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) 2015-2020 Clearmatics Technologies Ltd
+//
+// SPDX-License-Identifier: LGPL-3.0+
+
+#ifndef __ZECALE_CIRCUITS_FIELDS_FP6_3OVER2_GADGETS_HPP__
+#define __ZECALE_CIRCUITS_FIELDS_FP6_3OVER2_GADGETS_HPP__
+
+#include <libff/algebra/curves/public_params.hpp>
+#include <libsnark/gadgetlib1/gadget.hpp>
+#include <libsnark/gadgetlib1/gadgets/fields/fp2_gadgets.hpp>
+
+namespace libzecale
+{
+
+template<typename Fp6T>
+class Fp6_3over2_variable : public libsnark::gadget<typename Fp6T::my_Fp>
+{
+public:
+    using FieldT = typename Fp6T::my_Fp;
+    using Fp2T = typename Fp6T::my_Fp2;
+
+    libsnark::Fp2_variable<Fp2T> _c0;
+    libsnark::Fp2_variable<Fp2T> _c1;
+    libsnark::Fp2_variable<Fp2T> _c2;
+
+    Fp6_3over2_variable(
+        libsnark::protoboard<FieldT> &pb, const std::string &annotation_prefix);
+
+    Fp6_3over2_variable(
+        libsnark::protoboard<FieldT> &pb,
+        const Fp6_3over2_variable<Fp6T> &v,
+        const std::string &annotation_prefix);
+
+    Fp6_3over2_variable(
+        libsnark::protoboard<FieldT> &pb,
+        const Fp6T &v,
+        const std::string &annotation_prefix);
+
+    Fp6_3over2_variable(
+        libsnark::protoboard<FieldT> &pb,
+        const libsnark::Fp2_variable<Fp2T> &c0,
+        const libsnark::Fp2_variable<Fp2T> &c1,
+        const libsnark::Fp2_variable<Fp2T> &c2,
+        const std::string &annotation_prefix);
+
+    Fp6_3over2_variable<Fp6T> operator*(const FieldT &other);
```

Comment:
This is not consistent with your comment here: https://github.com/clearmatics/libsnark/pull/10#discussion_r449134158

what about switching `other` for `scalar` as suggested here: https://github.com/clearmatics/zecale/pull/17#discussion_r450915351 ?

Reply:
renamed